flink-scala/src/main/scala/org/apache/flink/api/scala/typeutils/OptionSerializer.scala

```diff
@@ -65,7 +65,7 @@ class OptionSerializer[A](val elemSerializer: TypeSerializer[A])
     case Some(a) =>
       target.writeBoolean(true)
       elemSerializer.serialize(a, target)
-    case None =>
+    case _ =>
```

Comment:
I understand. Is there possible this case as serialization with new serializer and deserialization with old one? Exception will be in this case, isn't it?